### PR TITLE
Add Auth class and user related functions.

### PR DESCRIPTION
**Pull request by liayoo**
- getUser
- addUserEthAddress
- removeUserEthAddress

### CODE REVIEW

**Review by platfowner — APPROVED** [review]
LGTM
Thanks!

File: src/auth.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      appId,
+      userId,
+      timestamp: Date.now(),
+    };
+    const signature = this.ain.wallet.sign(stringify(data));
+    return axios
+      .get(`${this.baseUrl}/user`, {
+        data: { data, signature },
+      })
+      .then((res) => res.data)
```
Comment (woomurf) [inline src/auth.ts:30]: How about res.data.data?  
res.data like this 
```
{ code, message, data }
```

File: src/auth.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    const data = {
+      appId,
+      userId,
+      ethAddress,
+      timestamp: Date.now(),
+    };
+    const signature = this.ain.wallet.sign(stringify(data));
+    return axios
+      .post(`${this.baseUrl}/user/ethAddress`, { data, signature })
+      .then((res) => res.data)
```
Comment (woomurf) [inline src/auth.ts:46]: ditto

File: src/auth.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      userId,
+      ethAddress,
+      timestamp: Date.now(),
+    };
+    const signature = this.ain.wallet.sign(stringify(data));
+    return axios
+      .delete(`${this.baseUrl}/user/ethAddress`, {
+        data: { data, signature },
+      })
+      .then((res) => res.data)
```
Comment (woomurf) [inline src/auth.ts:64]: ditto

File: src/auth.ts (hunk 1 above)
Comment (liayoo) [reply src/auth.ts]: right! fixed!

File: src/auth.ts (hunk 2 above)
Comment (liayoo) [reply src/auth.ts]: ditto

File: src/auth.ts (hunk 3 above)
Comment (liayoo) [reply src/auth.ts]: ditto

**Comment by liayoo** [comment]
Thanks for the review, now merging~

**Pull request merged**